Course/Course/Presentation/Container/CourseContainerViewModel.swift

```diff
@@ -431,6 +474,36 @@ public class CourseContainerViewModel: BaseCourseViewModel {
         }
         .store(in: &cancellables)
     }
+    
+    deinit {
+        let center = NotificationCenter.default
+        guard let shiftCourseDatesObserver = self.shiftCourseDatesObserver else { return }
+        center.removeObserver(shiftCourseDatesObserver)
+    }
+}
+
+extension CourseContainerViewModel {
+    func addNotificationObserver() {
+        let center = NotificationCenter.default
+        shiftCourseDatesObserver = center.addObserver(
+            forName: .shiftCourseDates,
+            object: nil,
+            queue: nil
+        ) { [weak self] notification in
+            if let courseID = notification.object as? String {
+                guard let strongSelf = self else {
+                    return
+                }
+                Task {
+                    await strongSelf.getCourseBlocks(courseID: courseID, withProgress: true)
+                    await strongSelf.getCourseDeadlineInfo(courseID: courseID, withProgress: true)
+                    DispatchQueue.main.async {
+                        strongSelf.dueDatesShifted = true
+                    }
```

Comment:
How about defining `addNotificationObserver` as `MainActor` so we don't have to define Task here and even can omit `DispatchQueue`